File: core/src/commonMain/kotlin/com/xhsdn/core/url/UrlUtils.kt
package com.xhsdn.core.url

/**
 * 通用 URL 工具。从 [app/.../utils/UrlUtils.kt] 迁出，
 * 保持与 Android 端原 [com.neoruaa.xhsdn.utils.UrlUtils] 同名 API，调用方零改动。
 */
object UrlUtils {
    private val URL_REGEX = Regex("(?:https?|ftp)://[^\\s]+")

    /** 从一段文本中抽取第一个看起来像 URL 的字符串。 */
    fun extractFirstUrl(text: String?): String? {
        if (text.isNullOrEmpty()) return null
        val match = URL_REGEX.find(text) ?: return null
        return match.value
    }

    /** 是否为 XHS 链接（任何已知格式）。 */
    fun isXhsLink(url: String?): Boolean {
        if (url.isNullOrEmpty()) return false
        return LinkExtractor.extractLinks(url).isNotEmpty() ||
            url.contains("xiaohongshu.com") ||
            url.contains("xhslink.com")
    }
}

File: web/src/core/url/UrlUtils.ts
import { extractLinks } from './LinkExtractor';

/**
 * 通用 URL 工具。1:1 翻译自
 * `core/src/commonMain/kotlin/com/xhsdn/core/url/UrlUtils.kt`。
 */
const URL_REGEX = /(?:https?|ftp):\/\/[^\s]+/;

/** 从一段文本中抽取第一个看起来像 URL 的字符串。 */
export function extractFirstUrl(text: string | null | undefined): string | null {
  if (!text) return null;
  const match = text.match(URL_REGEX);
  return match ? match[0] : null;
}

/** 是否为 XHS 链接(任何已知格式)。 */
export function isXhsLink(url: string | null | undefined): boolean {
  if (!url) return false;
  if (extractLinks(url).length > 0) return true;
  return url.includes('xiaohongshu.com') || url.includes('xhslink.com');
}
